### PR TITLE
CSP default directives

### PR DESCRIPTION
**Pull request by thomasdingemanse**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

This is my pull request for issue #108. Should I add documentation? I don't think it's necessary in this case, since the README already says:
> It accepts the same options of Helmet, and you can see more in [the helmet documentation](https://helmetjs.github.io/docs/).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,15 +3,19 @@
...
 
   app.addHook('onRequest', function (req, reply, next) {
     middleware(req.raw, reply.raw, next)
   })
 
   next()
-}, {
+}
+
+fastifyHelmet.contentSecurityPolicy = helmet.contentSecurityPolicy
```
Comment (mcollina) [inline index.js:16]: Can you set it as `module.exports.contentSecurityPolicy` instead? That'd make Node esm named exports work.

Comment (thomasdingemanse) [reply index.js]: Yes, thanks for the suggestion, that's better.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by thomasdingemanse** [comment]
Wait, I've found an issue with this. Adding additional directives mysteriously only works when they are provided in kebab case. This works...

```
app.register(helmet, {
  contentSecurityPolicy: {
    directives: {
      ...fastifyHelmet.contentSecurityPolicy.getDefaultDirectives(),
      'script-src': ['\'self\'', 'http://localhost:3000']
    }
  }
})
```

... while this does not:

```
app.register(helmet, {
  contentSecurityPolicy: {
    directives: {
      ...fastifyHelmet.contentSecurityPolicy.getDefaultDirectives(),
      scriptSrc: ['\'self\'', 'http://localhost:3000']
    }
  }
})
```

I first thought it didn't work because `getDefaultDirectives()` is called after the contentSecurityPolicy middleware has been initialized in fastify, whereas in the original helmet, the options are passed to the first call to `helmet()` in an options parameter:

```
app.use(
  helmet.contentSecurityPolicy({
    directives: {
      ...helmet.contentSecurityPolicy.getDefaultDirectives(),
      "script-src": ["'self'", "example.com"],
    },
  })
);
```

But that does not explain why it does work in kebab case and not in camel case. Maybe this is just the CSP middleware being weird? At least the default directives can be used now...

**Comment by thomasdingemanse** [comment]
I don't have any experience with typescript, but I'll try to figure out why this happens.

**Comment by thomasdingemanse** [comment]
Turns out the exact same thing happens with Express and the original helmet:

```js
const express = require('express')
const helmet = require('helmet')

const app = express()

const PORT = 3000

app.use(helmet.contentSecurityPolicy({
    directives: {
        ...helmet.contentSecurityPolicy.getDefaultDirectives(),
        scriptSrc: ["'self'", "example.com"]
    }
}))

app.get('/', (req, res) => {
    res.send('Hello World!')
})

app.listen(port, () => {
    console.log(`Example app listening at http://localhost:${PORT}`)
})
```

```
            throw new Error(`Content-Security-Policy received a duplicate directive ${JSON.stringify(directiveName)}`);
            ^

Error: Content-Security-Policy received a duplicate directive "script-src"
```

I'm not sure if this is the intended behavior or not, but the issue is in the CSP middleware itself, not fastify-helmet. I'll open an issue there. I won't be adding commits to this pull request anymore (unless anything else comes up).

**Comment by Fdawgs** [comment]
How did you get on with raising an issue @thomasdingemanse?
Ran into the same problem myself!

**Comment by mcollina** [comment]
@thomasdingemanse should this be landed?

**Comment by thomasdingemanse** [comment]
@Fdawgs Oops, sorry, I forgot. I'll do it today!

**Comment by thomasdingemanse** [comment]
@mcollina With this PR, it's possible to get the default directives, so in my opinion the PR can be merged.

The remaining issue that directives can only be added in kebab-case should be fixed in `helmet` itself but I don't think it should require any additional changes to `fastify-helmet` apart from changing the `helmet` dependency version afterwards.

**Comment by thomasdingemanse** [comment]
I've created an issue: https://github.com/helmetjs/helmet/issues/289#issue-796787354

**Pull request merged**